libraries/exoplayer/src/main/java/androidx/media3/exoplayer/drm/HttpMediaDrmCallback.java

```diff
@@ -182,7 +192,17 @@ private static byte[] executePost(
       while (true) {
         DataSourceInputStream inputStream = new DataSourceInputStream(dataSource, dataSpec);
         try {
-          return Util.toByteArray(inputStream);
+          byte[] response = Util.toByteArray(inputStream);
+          lastLoadEventInfo =
+              new LoadEventInfo(
+                  -1,   // note this is replaced with the actual taskId from the request
+                  originalDataSpec,
+                  dataSource.getLastOpenedUri(),
+                  dataSource.getLastResponseHeaders(),
+                  SystemClock.elapsedRealtime(),
+                  /* loadDurationMs= */ SystemClock.elapsedRealtime() - startTimeMs,
+                  ((byte []) response).length);
+            return response;
```

Comment:
Creating the `LoadEventInfo` in the actual `HttpMediaDrmCallback` addresses all the todo's with its correctness issues.   Also note now the `loadDurationMs` does not include any queuing time in the `RequestHandler` (so it's pure network and proxy overhead measured).

Lastly, `DataSource.getLastOpenedUri()` should get the redirected URI (if any), at least this is the contract for `DataSource.getUri()`

Reply:
Looks generally good, thanks